### ISSUE TITLE
rush deploy fails when package already "deployed"

### ISSUE THREAD

**Issue by btilford** [microsoft/rushstack#2121]
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [ ] Feature
- [x] Bug

**Please describe the actual behavior.**

Running `rush deploy --project xyz` outputs 
```
ERROR: '/repo/common/deploy/lib/lib-a/dist/index.js' already exists
```

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

I _think_ this happens when you have more than one package in the graph that have the same dependency.

* libs/lib-a
* libs/lib-b 
    * **dependencies**
         * lib-a
* apps/my-app -- item being "deployed"
     * **dependencies**
          * lib-a
          * lib-b

     
    

**What is the expected behavior?**

When resolving the dependency graph **and** the package is part of the current monorepo it should only show up once in the dependency graph **or** can be overwritten without an error. 

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** 
Rush Multi-Project Build Tool 5.30.1 - https://rushjs.io
Node.js version is 12.16.3 (LTS)

**Comment by octogonz** [comment]
@btilford Would you be able to provide a small branch that repros this error? That would help us fix it faster.

**Comment by btilford** [comment]
@octogonz https://github.com/btilford/rush-deploy-2121

I did find out it has something to do with the **files** array in package.json. If I remove that it seems to be fine but then on a real project I get all sorts of stuff I don't want published in a lib or deployed in an app. iirc there are like 100 ways to include / exclude files with NPM so I can look into other options there.

**Comment by octogonz** [comment]
> I did find out it has something to do with the **files** array in package.json.

We generally consider it not a good practice to put build settings in **package.json** because:
1. That file has no schema validation, e.g. if you misspelled `"files"` it would be silently ignored with no idea why it's not working
2. The NPM protocol sometimes downloads every package.json from every published version of a package, so every extra setting makes this payload larger and installation slower.

That said, `rush deploy` should handle this or report an error saying it's not supported. The error message you reported isn't very understandable.

> https://github.com/btilford/rush-deploy-2121

@btilford  I tried this:
1. Clone your `master` branch
2. Run `rush install`
3. Run `rush deploy --overwrite`

It deployed successfully without reporting any error. Am I doing something wrong?

**Comment by btilford** [comment]
@octogonz Sorry should have added the dist/ directory. It's created if you run `rush build`

Is there a preferred way to include files in a package `.npmignore` etc? **files** was a bit more strict and prevented accidentally including things since I didn't have to list every item to ignore.

**Comment by octogonz** [comment]
> **files** was a bit more strict and prevented accidentally including things since I didn't have to list every item to ignore.

We use a negative ignore pattern like this: [node-core-library/.npmignore](https://github.com/microsoft/rushstack/blob/master/libraries/node-core-library/.npmignore)

**Comment by octogonz** [comment]
I was able to repro it. It seems to be a bug in the `npm-packlist` library that builds the inventory of filenames that `npm pack` would have published.  When it sees this in **package.json**:
```
  "files": [
    "dist/"
  ],
```
...it returns this inventory:
```
'dist//index.js'
'dist/index.js'
'package.json'
```
Whereas if you change it to `"dist"` then it returns the correct result:
```
'dist/index.js'
'package.json'
```

Seems like `rush deploy` needs to normalize and dedupe the file paths before copying.

**Comment by octogonz** [comment]
Here's a PR: https://github.com/microsoft/rushstack/pull/2126

@btilford thank you for taking the time to prepare a repro -- it made this easy to debug!

**Comment by octogonz** [comment]
This fix was released with Rush 5.33.1
